apps/trading/pages/markets/trade-grid.tsx

```diff
@@ -31,57 +34,110 @@ interface TradeGridProps {
   market: Market_market;
 }
 
+export const TradeMarketHeader = ({ market }: TradeGridProps) => {
+  const [open, setOpen] = useState(false);
+  const candlesClose: string[] = (market?.candles || [])
+    .map((candle) => candle?.close)
+    .filter((c): c is CandleClose => c !== null);
+  const headerItemClassName = 'whitespace-nowrap flex flex-col';
+  const itemClassName =
+    'font-sans font-normal mb-0 text-dark/80 dark:text-white/80 text-ui-small';
+  const itemValueClassName =
+    'capitalize font-sans tracking-tighter text-black dark:text-white text-ui';
+  return (
+    <header className="w-full p-8">
+      <SelectMarketDialog dialogOpen={open} setDialogOpen={setOpen} />
+      <div className="flex flex-col md:flex-row gap-20 md:gap-64 ml-auto mr-8">
+        <button
+          onClick={() => setOpen(!open)}
+          className="shrink-0 dark:text-vega-yellow text-black text-h5 flex items-center gap-8 px-4 py-0 h-37 hover:bg-vega-yellow dark:hover:bg-white/20"
+        >
+          <span className="break-words text-left">{market.name}</span>
+          <ArrowDown color="yellow" borderX={8} borderTop={12} />
+        </button>
+
+        <div className="flex flex-auto items-start gap-64 overflow-x-scroll whitespace-nowrap w-[400px]">
+          <div className={headerItemClassName}>
+            <span className={itemClassName}>Change (24h)</span>
+            {
+              <PriceCellChange
+                candles={candlesClose}
+                decimalPlaces={market.decimalPlaces}
+              />
+            }
+          </div>
+          <div className={headerItemClassName}>
+            <span className={itemClassName}>Volume</span>
+            <span className={itemValueClassName}>
+              {market.data?.indicativeVolume === '0'
```

Comment:
question (non-blocking): not how the data since the data is an optional property, it feels like it would be appropriate to render a hyphen for that case, too. not sure, if this is not a real scenario, probably doesn't matter that much.

Reply:
makes sense, & updated both in my last commits